Library/Homebrew/cmd/audit.rb

```diff
@@ -23,65 +26,51 @@
 require "date"
 
 module Homebrew
+  RUBY_2_OR_LATER = RUBY_VERSION.split(".").first.to_i >= 2
+
   def audit
     formula_count = 0
     problem_count = 0
 
     strict = ARGV.include? "--strict"
-    if strict && ARGV.resolved_formulae.any? && RUBY_VERSION.split(".").first.to_i >= 2
-      require "cmd/style"
-      ohai "brew style #{ARGV.resolved_formulae.join " "}"
-      style
-    end
-
+    style = strict && ARGV.resolved_formulae.any? && RUBY_2_OR_LATER
     online = ARGV.include? "--online"
 
     ENV.activate_extensions!
     ENV.setup_build_environment
 
-    if ARGV.switch? "D"
-      FormulaAuditor.module_eval do
-        instance_methods.grep(/audit_/).map do |name|
-          method = instance_method(name)
-          define_method(name) do |*args, &block|
-            begin
-              time = Time.now
-              method.bind(self).call(*args, &block)
-            ensure
-              $times[name] ||= 0
-              $times[name] += Time.now - time
-            end
-          end
-        end
-      end
-
-      $times = {}
-      at_exit { puts $times.sort_by { |_k, v| v }.map { |k, v| "#{k}: #{v}" } }
-    end
-
-    ff = if ARGV.named.empty?
-      Formula
+    if ARGV.named.empty?
+      ff = Formula
+      files = [HOMEBREW_LIBRARY]
     else
-      ARGV.resolved_formulae
+      ff = ARGV.resolved_formulae
+      files = ARGV.resolved_formulae.map(&:path)
+    end
+    if style
+      require "cmd/style"
+      # Check style in a single batch run up front for performance
+      style_results = check_style_json(files, :realpath => true)
     end
 
-    output_header = !strict
+    output_header = false
 
     ff.each do |f|
-      fa = FormulaAuditor.new(f, :strict => strict, :online => online)
+      options = { :strict => strict, :online => online }
+      options[:style_offenses] = style_results.file_offenses(f.path) if style
+      fa = FormulaAuditor.new(f, options)
       fa.audit
 
-      unless fa.problems.empty?
-        unless output_header
-          puts
-          ohai "audit problems"
-          output_header = true
-        end
+      next if fa.problems.empty?
 
-        formula_count += 1
-        problem_count += fa.problems.size
-        puts "#{f.full_name}:", fa.problems.map { |p| " * #{p}" }, ""
+      unless output_header
+        ohai "audit problems"
+        puts
+        output_header = true
       end
```

Comment:
I think we can completely eliminate the `unless output_header` block. It only existed in the past because if you passed `--strict` it would be difficult to distinguish between when the output of `brew style` ends and the output of `brew audit` begins. Now that the style violations are folded into the list of audit problems, this header is no longer necessary. Just like with `brew audit` (without `--strict`) before this PR.